### PR TITLE
Add missing shell script and CI

### PR DESCRIPTION
**Pull request by xuhang57**
This: https://github.com/CCI-MOC/M2/blob/master/ci/pycodestyle_tests.sh
is actually looking for a shell script called: `list_tracked_pyfiles.sh`

This patch adds the missing file and copying from the HIL codebase. Put the link in the comment.

### CODE REVIEW

File: ci/list_tracked_pyfiles.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+# https://github.com/CCI-MOC/hil/blob/master/ci/list_tracked_pyfiles.sh
```
Comment (chemistry-sourabh) [inline ci/list_tracked_pyfiles.sh]: Can you add a line saying what is the objective of this shell script ?

**Comment by chemistry-sourabh** [comment]
Also another question, I found this line in our current travis.yml
```find . -name \*.py -exec pep8 --ignore=E402 {} +```

Why can't we just replace pep8 in this line with pycodestyle ?

**Comment by xuhang57** [comment]
let's get this reviewed and merged so we can have a working Travis CI. Thanks for reviewing @chemistry-sourabh @naved001

File: ci/list_tracked_pyfiles.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+# This script lists all of the python source files that are tracked
+# by git. The primary use is deciding what files to run the linters on.
+# https://github.com/CCI-MOC/hil/blob/master/ci/list_tracked_pyfiles.sh
```
Comment (djfinn14) [inline ci/list_tracked_pyfiles.sh]: Should this be CCI-MOC/M2 instead of CCI-MOC/hil ?

**Comment by naved001** [comment]
> Also another question, I found this line in our current travis.yml
find . -name \*.py -exec pep8 --ignore=E402 {} +

Why can't we just replace pep8 in this line with pycodestyle ?

File: model/drivers/interfaces/ipxe.py [hunk 3]
```diff
@@ -3,21 +3,14 @@
 
 class ipxe:
```
Comment (djfinn14) [inline model/drivers/interfaces/ipxe.py]: I believe it is coding convention to use CapWords convention for naming classes, not sure if it should be Ipxe or IPXE, though.

File: ci/list_tracked_pyfiles.sh (hunk 2 above)
Comment (xuhang57) [reply ci/list_tracked_pyfiles.sh]: darn. thanks!

File: model/drivers/interfaces/ipxe.py (hunk 3 above)
Comment (xuhang57) [reply model/drivers/interfaces/ipxe.py]: @djfinn14 good point. Sourabh is cleaning this up in his PR. so this PR would just focus on the CI then. Thanks!

File: ci/list_tracked_pyfiles.sh (hunk 2 above)
Comment (xuhang57) [reply ci/list_tracked_pyfiles.sh]: this was a reference since I literally just copy and pasted this file from HIL. delete the link.

**Comment by xuhang57** [comment]
@chemistry-sourabh @naved001 @djfinn14 All done. 👍

**Comment by xuhang57** [comment]
I think i should put the pylint code along with this PR then @naved001 

https://github.com/CCI-MOC/M2/issues/15

**Comment by xuhang57** [comment]
@chemistry-sourabh Yup. But feel free to change that in your PR. I just modified it for passing the CI.

As for that, I was discussed with Dan earlier today. I grabbed it from the previous BMI Travis and not sure whether or not need it. What do you think? If not, then I will take it out.

**Comment by chemistry-sourabh** [comment]
@naved001 we I remember right I think pylint is doing the job of pycompile and pycodestyles right ?

**Comment by naved001** [comment]
> I think pylint is doing the job of pycompile and pycodestyles right ?

The way we are running pylint right now, we disable all checks and enable only the ones that we like. I think we will have to find the flags we want to enable for syntax checks.

If you just run pylint raw it will look for syntax errors and every other error in the universe making it noisy.

**Comment by xuhang57** [comment]
@naved001 done.

**Pull request merged**